frontends/main/src/app-pages/DashboardPage/CoursewareDisplay/DashboardCard.tsx

```diff
@@ -308,15 +310,20 @@ const DashboardCard: React.FC<DashboardCardProps> = ({
   )
 
   const mobileLayout = (
-    <CardRoot data-testid="enrollment-card-mobile">
+    <CardRoot
+      screenSize="mobile"
+      data-testid="enrollment-card-mobile"
+      as={Component}
+      className={className}
+    >
       <Stack
         direction="row"
         justifyContent="space-between"
         alignItems="stretch"
         flex={1}
         width="100%"
       >
-        <Stack direction="column" gap="8px">
+        <Stack direction="column" gap="8px" flex={1}>
```

Comment:
This is to address an odd issue i noticed on the mobile cards.

Before:

![Screenshot 2025-04-09 at 3 58 41 PM](https://github.com/user-attachments/assets/42d074f8-c314-4540-82de-0d17bfb2fcdc)


Now:

![Screenshot 2025-04-09 at 3 58 55 PM](https://github.com/user-attachments/assets/a462b2e1-8fc2-4d86-90b4-0033c6defb21)


I don't entirely understand why the "Before" behavior was wrapping that way, though.